### PR TITLE
Add distrib. arch. guide intro and autoscaling basics

### PR DESCRIPTION
**Pull request by DiannaHohensee**

### CODE REVIEW

File: docs/internal/DistributedArchitectureGuide.md [hunk 1]
```diff
@@ -254,9 +261,104 @@ works in parallel with the storage engine.)
...
+to search load. Paired with ES Cloud, autoscaling only scales upward, not downward, except for ML nodes
+that do get scaled up _and_ down.
+
+The external autoscaler, external to ES, can be found in the [elasticsearch-autoscaler][] repository.
+
+[based on user defined policies]: https://www.elastic.co/guide/en/elasticsearch/reference/current/xpack-autoscaling.html
+[deciders]: https://www.elastic.co/guide/en/elasticsearch/reference/current/autoscaling-deciders.html
+[webinar on autoscaling]: https://www.elastic.co/webinars/autoscaling-from-zero-to-production-seamlessly
+[Autoscaling APIs]: https://www.elastic.co/guide/en/elasticsearch/reference/current/autoscaling-apis.html
+[elasticsearch-autoscaler]: https://github.com/elastic/elasticsearch-autoscaler
```
Comment (fcofdez) [inline docs/internal/DistributedArchitectureGuide.md:283]: This is the autoscaler for serverless, which uses a different set of APIs and concepts.

Comment (DiannaHohensee) [reply docs/internal/DistributedArchitectureGuide.md]: Removed. Thanks for taking a look and catching that

File: docs/internal/DistributedArchitectureGuide.md [hunk 2]
```diff
@@ -254,9 +263,101 @@ works in parallel with the storage engine.)
...
+[TransportGetAutoscalingCapacityAction]: https://github.com/elastic/elasticsearch/blob/8.10/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/action/TransportGetAutoscalingCapacityAction.java#L81-L97
+
+### How cluster capacity is determined
+
+[AutoscalingMetadata][] implements [Metadata.Custom][] in order to persist autoscaling policies. Each
+Decider is an implementation of [AutoscalingDeciderService][]. The [AutoscalingCalculateCapacityService][]
+is responsible for running the calculation.
+
+[TransportGetAutoscalingCapacityAction.computeCapacity] is the entry point to [AutoscalingCalculateCapacityService.calculate],
+which creates a [AutoscalingDeciderResults][] for [each autoscaling policy][]. [AutoscalingDeciderResults.toXContent][] then
```
Comment (pxsalehi) [inline docs/internal/DistributedArchitectureGuide.md:306]: Aren't the capacity calculation done per decider per policy?

File: docs/internal/DistributedArchitectureGuide.md [hunk 3]
```diff
@@ -254,9 +263,101 @@ works in parallel with the storage engine.)
...
+[DiskThresholdMonitor][] is one such data source. The Monitor runs on the master node and maintains
+lists of nodes that exceed various disk size thresholds. [DiskThresholdSettings][] contains the
+threshold settings with which the `DiskThresholdMonitor` runs.
+
+[DiskThresholdMonitor]: https://github.com/elastic/elasticsearch/blob/8.10/server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java#L52-L57
+[DiskThresholdSettings]: https://github.com/elastic/elasticsearch/blob/8.10/server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdSettings.java#L24-L27
+
+### Deciders
+
+The `ReactiveStorageDeciderService` uses tracks information that demonstrates storage limitations are causing
```
Comment (pxsalehi) [inline docs/internal/DistributedArchitectureGuide.md:344]: uses or tracks?

File: docs/internal/DistributedArchitectureGuide.md [hunk 4]
```diff
@@ -254,9 +263,101 @@ works in parallel with the storage engine.)
...
+
+The `ProactiveStorageDeciderService` maintains a forecast window that [defaults to 30 minutes][]. It only
+runs on data streams (ILM, rollover, etc), not regular indexes. It looks at past [index changes][] that
+took place within the forecast window to [predict][] resources that will be needed shortly.
+
+[defaults to 30 minutes]: https://github.com/elastic/elasticsearch/blob/8.10/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ProactiveStorageDeciderService.java#L32
+[index changes]: https://github.com/elastic/elasticsearch/blob/8.13/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ProactiveStorageDeciderService.java#L79-L82
+[predict]: https://github.com/elastic/elasticsearch/blob/8.13/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ProactiveStorageDeciderService.java#L85-L95
+
+There are several more Decider Services: grep for `implements AutoscalingDeciderService`.
```
Comment (pxsalehi) [inline docs/internal/DistributedArchitectureGuide.md:360]: maybe instead of grep, just pointing to the list of deciders in the docs is enough.

**Review by pxsalehi — COMMENTED** [review]
Thanks, Dianna. Maybe @henningandersen also could take a quick look?

File: docs/internal/DistributedArchitectureGuide.md (hunk 2 above)
Comment (DiannaHohensee) [reply docs/internal/DistributedArchitectureGuide.md:289]: Ummm.. Pretty hard to tell that it isn't per policy. The [code strongly suggests per policy](https://github.com/elastic/elasticsearch/blob/8.10/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/capacity/AutoscalingCalculateCapacityService.java#L118) -- whatever [AutoscalingMetadata holds](https://github.com/elastic/elasticsearch/blob/8.10/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/AutoscalingMetadata.java#L67-L71). This is also into the weeds of the code. Perhaps the results are grouped per decider later on in the process.

File: docs/internal/DistributedArchitectureGuide.md [hunk 5]
```diff
@@ -1,6 +1,15 @@
...
 
-(Summary, brief discussion of our features)
+The Distributed Area is split into Coordination and Indexing.
+
+Indexing covers the index path, stretching from the user REST command through shard routing down to the translog and storage
+engine. It includes reindexing: copying from a source index to a destination index. Shard recovery and cross cluster replication
+also fall under indexing.
+
+Coordination encompasses the distributed components relating to cluster coordination, networking, shard allocation decisions
+(i.e. the balancer algorithms), cluster autoscaling stats, the discovery plugin system, task management, and snapshot/restore.
```
[the remark this answers is not in the document]
Comment (DiannaHohensee) [reply docs/internal/DistributedArchitectureGuide.md]: Could you clarify what you mean by headlines? Also, [this was the original change](https://github.com/elastic/elasticsearch/pull/107284/commits/91462448321de9cca7f899085d3c0091090402e7), is roughly that format preferable?

File: docs/internal/DistributedArchitectureGuide.md (hunk 2 above)
Comment (pxsalehi) [reply docs/internal/DistributedArchitectureGuide.md:289]: Yeah, inside [`calculateForPolicy`](https://github.com/elastic/elasticsearch/blob/782234a84d8aebda5c5b9cd1b74138f03ba36504/x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/capacity/AutoscalingCalculateCapacityService.java#L165) we calculate per decider, it seems.

File: docs/internal/DistributedArchitectureGuide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (pxsalehi) [reply docs/internal/DistributedArchitectureGuide.md]: I meant just listing the topics (each seem to be one headline) is probably enough. The team responsibility areas might not be important there. But do it as you prefer.

File: docs/internal/DistributedArchitectureGuide.md [hunk 6]
```diff
@@ -254,9 +263,101 @@ works in parallel with the storage engine.)
 
 # Autoscaling
 
-(Reactive and proactive autoscaling. Explain that we surface recommendations, how control plane uses it.)
-
-(Sketch / list the different deciders that we have, and then also how we use information from each to make a recommendation.)
+Autoscaling in ES (Elasticsearch) is not an active process that moves data on its own initiative, but
+rather a recommendation system using cluster and node level statistics. ES Autoscaling is paired with
+an ES Cloud service that periodically polls the ES elected master node for suggested cluster changes.
+The cloud service will add more servers/nodes, or increase the resources of existing nodes.
```
[the remark this answers is not in the document]
Comment (pxsalehi) [reply docs/internal/DistributedArchitectureGuide.md]: Lol! *vertical autoscaling*! Not sure how i typed that instead! :))))

**Review by pxsalehi — APPROVED** [review]

File: docs/internal/DistributedArchitectureGuide.md [hunk 7]
```diff
@@ -1,6 +1,14 @@
-# Distributed Area Team Internals
+# Distributed Area Internals
 
-(Summary, brief discussion of our features)
+The Distributed Area contains indexing and coordination systems.
+
+The index path stretches from the user REST command through shard routing down to each individual shard's translog and storage
+engine. Reindexing is effectively reading from a source index and writing to a destination index (perhaps on different nodes).
+The coordination side includes cluster coordination, shard allocation, cluster autoscaling stats, task management, and cross
+cluster replication. Less obviously coordination systems include networking, the discovery plugin system, the snapshot/restore
```
Comment (DiannaHohensee) [inline docs/internal/DistributedArchitectureGuide.md:8]:
```suggestion
cluster replication. Less obvious coordination systems include networking, the discovery plugin system, the snapshot/restore
```

**Pull request merged**